### PR TITLE
[core] feat: SwitchCard component

### PR DESCRIPTION
**Pull request by CPerinet**
~Based on https://github.com/palantir/blueprint/pull/6246~ edit: that PR has merged

Screenshot 2023-06-27 at 12 57 28 PM

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204971665339519

### CODE REVIEW

**Comment by adidahiya** [comment]
I fixed the double click issue by using the switch `onChange` handler instead of a new `onClick` on the containing Card. This required moving the card padding down to the label element.

**Comment by adidahiya** [comment]
I fixed the issue where the right side of the card was non-interactive:

**Comment by adidahiya** [comment]
[latest docs preview build](https://output.circle-artifacts.com/output/job/f5d34f6a-eae5-42ec-be40-3904e5f7cce4/artifacts/0/packages/docs-app/dist/index.html#core/components/card.switch-card)

**Comment by CPerinet** [comment]
- The padding on card list cards does not get overriden correctly.
- Do we want to also add radio and checkbox cards?

**Comment by adidahiya** [comment]
> The padding on card list cards does not get overriden correctly.

Ah, I see the issue. I'll push a fix:




> Do we want to also add radio and checkbox cards?

Yes, in a separate PR

File: packages/docs-app/src/examples/core-examples/switchCardExample.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ * limitations under the License.
+ */
+
+import * as React from "react";
+
+import { CardList, H5, Switch, SwitchCard } from "@blueprintjs/core";
+import { Example, ExampleProps, handleBooleanChange } from "@blueprintjs/docs-theme";
+
+interface SwitchCardExampleState {
+    // TODO: add compact option
```
Comment (adidahiya) [inline packages/docs-app/src/examples/core-examples/switchCardExample.tsx:23]: @CPerinet we can add this option to this example once we implement `<Card compact={true} />` in a separate PR. The padding style variable has already been added in `_card-variables.scss` :+1:

**Comment by adidahiya** [comment]
I added an example showing composition with CardList:

image

**Review by adidahiya — APPROVED** [review]

**Pull request merged**